File: src/main/java/beemail/Message.java
package beemail;

import java.time.LocalDate;
import java.util.UUID;

public class Message {

    private String id = UUID.randomUUID().toString();
    private LocalDate timestamp = LocalDate.now();
    private String content;
    private User from;
    private User to;
    private boolean delivered = false;
    private boolean read = false;

    public Message(String content, User from, User to) {
        this.content = content;
        this.from = from;
        this.to = to;
    }

    public void log() {
        String format = "| %-15s | %-30s |%n";
        System.out.format("+-----------------+--------------------------------+%n");
        System.out.format("| Field           | Value                          |%n");
        System.out.format("+-----------------+--------------------------------+%n");
        System.out.format(format, "Created", timestamp);
        System.out.format(format, "From", from.getUserName());
        System.out.format(format, "To", to.getUserName());
        System.out.format(format, "Content", content);
        System.out.format(format, "Delivered", delivered);
        System.out.format(format, "Read", read);
        System.out.format("+-----------------+--------------------------------+%n");
    }

    public void markDelivered() {
        delivered = true;
    }

    public Message markRead() {
        read = true;
        return this;
    }
}

File: message.js
import { v4 as uuidv4 } from 'uuid';

// Message class
class Message {
    constructor(from, to, content) {
        this.id = uuidv4();
        this.datetime = new Date();
        this.content = content;
        this.from = from;
        this.to = to;
        this.delivered = false;
        this.read = false;
    }

    log() {
        console.log(`Message ID: ${this.id}, From: ${this.from.username}, To: ${this.to.username}, Sent: ${this.datetime}, Content: ${this.content}, Delivered: ${this.delivered}, Read: ${this.read}`);
    }

    markDelivered() {
        this.delivered = true;
    }

    markRead() {
        this.read = true;
    }
}

export default Message;